core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -799,9 +872,17 @@ private class BufferReleasingInputStream(
 
   override def close(): Unit = {
     if (!closed) {
-      delegate.close()
-      iterator.releaseCurrentResultBuffer()
-      closed = true
+      try {
+        delegate.close()
+        iterator.releaseCurrentResultBuffer()
+      } finally {
+        // Unset the flag when a remote request finished and free memory is fairly enough.
+        if (ShuffleBlockFetcherIterator.isNettyOOMOnShuffle.get &&
+            isNetworkReqDone && NettyUtils.freeDirectMemory() > iterator.averageRemoteBlockSize) {
```

Comment:
Nit: should the indentation be 2?

Reply:
it should be 4 indents for the if condition.